dns.go

```diff
@@ -104,30 +139,71 @@ func (api *API) DNSRecords(ctx context.Context, zoneID string, rr DNSRecord) ([]
 		v.Set("content", rr.Content)
 	}
 
+	if rr.Proxied != nil {
+		v.Set("proxied", strconv.FormatBool(*rr.Proxied))
```

Comment:
we don't need all of these `v.Set` calls as we can instead, pass it to `buildURI` and it will use the struct field configuration to build this out.

Reply:
This would require adding ``url:""`` to all the fields of `DNSRecord` as that is used for some fields when creating the URL. There is also the tag matching, which wouldn't fit inside the standard `buildURI` function.

Reply:
> This would require adding url:"" to all the fields of DNSRecord as that is used for some fields when creating the URL.

that's totally fine given it's only used when building query parameters.

> There is also the tag matching, which wouldn't fit inside the standard buildURI function.

i haven't looked at the API docs but this should be fine to add to the buildURI method or pass in a struct that allows nesting, etc.

Reply:
Docs sections

> DNS record tag value. For advanced queries, the following alternative  parameters are available: 'tag.\<tag-name\>.present',  'tag.\<tag-name\>.absent', 'tag.\<tag-name\>.exact',  'tag.\<tag-name\>.contains', 'tag.\<tag-name\>.startswith',  'tag.\<tag-name\>.endswith'. Just 'tag.\<tag-name\>' is  equivalent to 'tag.\<tag-name\>.exact'. Replace \<tag-name\>  with the name of the tag to query. Example:  /dns_records?tag.important.present&tag.team=DNS. Note that  'tag.\<tag-name\>.present' and 'tag.\<tag-name\>.absent' do not require a value; if one is given anyway it is ignored.

This means that you can have a URL parameter of `tag.example` or `tag.example.startswith` which is
https://github.com/cloudflare/cloudflare-go/pull/1151/files#diff-cb1b8a01bbc5f64cf06dce0e3f8021ad8289078b402dc620afa04780484d7cd5R152-R160
exists

Reply:
for simplicity sake, let's do all except for the tags. the others should be simple query parameters and we can look at incorporating this pattern later if it becomes a thing in more places.